File: 230114-dojo/java/app/src/main/java/Solution.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        Set<Integer> noDuplicatedItems = new LinkedHashSet<>();

        for (int number : nums) {
            noDuplicatedItems.add(number);
        }

        int options = noDuplicatedItems.size();
        int amountToGet = nums.length / 2;

        return Math.min(options, amountToGet);
    }
}

File: 230114-dojo/javascript/solution.js
export default function solution(nums) {
  const distictDuplicatedItems = new Set();

  nums.forEach((value) => distictDuplicatedItems.add(value));

  const options = distictDuplicatedItems.size;
  const availableAmount = nums.length / 2;

  return availableAmount >= options ? options : availableAmount;
}

// function solution(nums) {
//     const distictDuplicatedItems = new Set(nums);

//     const options = distictDuplicatedItems.size;
//     const availableAmount = nums.length / 2;

//     return availableAmount >= options ? options : availableAmount;
//   }
